crates/oxc_ast/src/ast/js.rs

```diff
@@ -343,9 +343,9 @@ impl<'a> PropertyKey<'a> {
             Self::PrivateIdentifier(_) => None,
             Self::Expression(expr) => match expr {
                 Expression::StringLiteral(lit) => Some(lit.value.clone()),
-                Expression::RegExpLiteral(lit) => Some(Atom::from(lit.regex.to_string())),
-                Expression::NumberLiteral(lit) => Some(Atom::from(lit.value.to_string())),
-                Expression::BigintLiteral(lit) => Some(Atom::from(lit.value.to_string())),
+                Expression::RegExpLiteral(lit) => Some(Atom::from(format!("{0}", lit.regex))),
```

Comment:
format still allocates, I think you can pass in a reference to Atom here?

Reply:
Oh ... so to_string is needed from regex and bigint, and not the others, my bad.